ext/CMakeLists.txt

```diff
@@ -1,2 +1,59 @@
+# Direct CMake submodules
 add_subdirectory(GViZDoom)
 add_subdirectory(json)
+
+# PyTorch requires a bit of extra work:
+
+# Create build directory
+set(PYTORCH_BUILD_DIR ${CMAKE_CURRENT_BINARY_DIR}/pytorch)
+file(MAKE_DIRECTORY ${PYTORCH_BUILD_DIR})
+
+# Directories for find_package
+set(LIBTORCH_DIR ${CMAKE_CURRENT_SOURCE_DIR}/pytorch/torch)
+set(Caffe2_DIR "${LIBTORCH_DIR}/share/cmake/Caffe2/")
+set(Torch_DIR "${PYTORCH_BUILD_DIR}/build/")
+set(ENV{TORCH_INSTALL_PREFIX} "${PYTORCH_BUILD_DIR}/build/")
+
+# Build libtorch in case it has not been built yet
+find_package(Torch QUIET)
+if (TORCH_FOUND)
+    message(STATUS "libtorch found, skipping the build")
+else()
+    message(STATUS "Building libtorch")
+    find_package(Python3 COMPONENTS Interpreter REQUIRED)
```

Comment:
remove

Reply:
This is the idiomatic way of finding python in CMake. In case you have multiple Python 3 versions installed, see: https://gitlab.kitware.com/cmake/cmake/-/issues/23139